### ISSUE TITLE
[FR/future PR] Firmware-based individual object cancelling

### ISSUE THREAD

**Issue by paukstelis** [MarlinFirmware/Marlin#15327]
I've been working on a Marlin feature that will allow cancelling of individual objects of a multi-object print. This leverages slicer features that places comments between blocks of gcode to delineate which gcode chunks belong to which object. This is used in the Octoprint-CancelObject plugin I developed. There would be a number of advantages having this cancelling feature available directly in firmware and I've made a working proof-of-concept that can be found here: https://github.com/paukstelis/Marlin/tree/cancelobjects

Here's a snippet of just some post-processing of a PrusaSlicer file with the Label Objects option turned to demonstrate how the blocks are set:

```
; first layer extrusion width = 0.42mm
O4; clear assigned objects
O1 I0; assign object
O1 I1; assign object
O1 I2; assign object
O1 I3; assign object
M201 X1000 Y1000 Z500 E10000 ; sets maximum accelerations, mm/sec^2
...
G1 X73.235 Y97.109 E0.05253
O2 I0 S1; start printing object 0
G1 E-2.30000 F2400.00000
G1 Z0.700 F6000.000
G1 X122.434 Y121.434
...
G1 X104.394 Y121.328 E0.03542
O2 I0 S0; stop printing object 0
O2 I1 S1; start printing object 1
G1 E-2.30000 F2400.00000
G1 Z0.700 F6000.000
```

Basically, I'm using these new O-codes to assign objects, set/unset active objects, and cancel objects. I'm looking for any feedback on how to make this more efficient with particular emphasis on working with the LCD (perhaps including object name strings  in the  menu) and any general coding help since I'm mostly learning on the fly. There are some limitations (must use relative extrusion distances) that will require checking that I haven't looked into yet.

**Comment by paukstelis** [comment]
This is addressed with https://github.com/MarlinFirmware/Marlin/commit/269700ff32f57a6acac88d9b1d1d08e9b05a037d

Closing.
